docs/migrations/support-slashed-denoms.md

```diff
@@ -0,0 +1,113 @@
+# Migrating from Unsupported Base Denom Slashes to Supporting Slashed BaseDenoms
+
+This document is intended to highlight significant changes which may require more information than presented in the CHANGELOG.
+Any changes that must be done by a user of ibc-go should be documented here.
+
+There are four sections based on the four potential user groups of this document:
+- Chains
+- IBC Apps
+- Relayers
+- IBC Light Clients
+
+**Note:** ibc-go supports golang semantic versioning and therefore all imports must be updated to bump the version number on major releases.
+```go
+github.com/cosmos/ibc-go/v3 -> github.com/cosmos/ibc-go/v4
+```
+
+This document is necessary when chains are upgrading from a version that does not support slashed base denoms (e.g. v3.0.0) to a version that does (e.g. v3.1.0).
```

Comment:
q: shouldn't this be v4.0.0 if you are bumping the go version?

Reply:
I believe the plan is to do a minor release and backport to v1 and v2. cc: @crodriguezvega

Reply:
https://github.com/cosmos/ibc-go/blob/main/RELEASES.md

Reply:
Yes, the idea is to release this fix in v3.1.0, but also as a minor backport in v2.3.0 and v1.5.0, so that chains not yet on v3 can still benefit from the fix. I think this is the fastest way to get this fix out. 

Alternatively, we could release this fix only with v4.0.0. That would probably make it easier for us to communicate this fix, because we can add this migration guide to the v3-to-v4 migration guide, and all chains migrating to v4.0.0 will need to run this migration. But v4.0.0 needs to come out with fee middleware and it could still take easily another 3 weeks before we cut the final release. So the viability of this option depends on how soon Evmos needs to have this fix.

Having a quick look at mintscan and it looks like most of the chains are still on v2, so if we release this only on v4 and those chains would like to benefit from the fix, then we are forcing them to run the migrations to upgrade to v3 and v4, which might be a bit too much to ask?

Does Evmos expect their slashed-denom tokens to be transferred to a particular set of chains or they could just go to any chain? Because if most likely the receiving chains of those tokens are going to be a handful of chains, maybe we could coordinate with an upgrade to v4?

@fedekunze @AdityaSripal what of these two options you think it's best?

Reply:
@crodriguezvega The idea is that slashed-denom tokens are supported on any cosmos chain and that there is no specific set of chains. 

As this functionality is blocking the adoption of our `x/erc20` module, it would be great to go for the first option and release it in v3.1.0, is it is the faster option and requires less coordination.

Reply:
Thanks for the feedback, @danburck! Yes, I think that it's the option that makes most sense. I will try to release is asap!